### DISCUSSION TITLE
How can I use `CyHttpMessages.IncomingHttpRequest` type in custom function?

### DISCUSSION THREAD

**Discussion by pleshevskiy** [cypress-io/cypress#25481] [Questions and Help]
I want to write a utility to wrap `cy.intercept` and use `CyHttpMessages.IncomingHttpRequest` for `req`, but for now I have to write `any`. It seems that `CyHttpMessage` is not exposed.

**Comment by portyu9** [comment]
Hi,

Yes. Import the network-stubbing namespace directly from Cypress's type bundle:

```ts
import type { CyHttpMessages } from 'cypress/types/net-stubbing'

export function handleRequest(
  req: CyHttpMessages.IncomingHttpRequest
) {
  // ...
}
```

You can also type the request and response bodies:

```ts
type RequestBody = {
  name: string
}

type ResponseBody = {
  id: string
}

type InterceptRequest =
  CyHttpMessages.IncomingHttpRequest<RequestBody, ResponseBody>
```

`CyHttpMessages` is exported from `cypress/types/net-stubbing`, so you do not need to leave `req` as `any` or try to access it as `Cypress.CyHttpMessages`.

Please mark this answer as accepted if it helped, thank you.
